### PR TITLE
VB EE: Properly handle syntax errors in synthesized assignments

### PR DESCRIPTION
**Pull request by tmat**
**Customer scenario**

The EE displays "Internal error in expression evaluator" when the value of local variable is set to an expression that's not syntactically correct in a Watch window.

**Bugs this fixes:**

https://github.com/dotnet/roslyn/issues/18531

**Workarounds, if any**

None. 

**Risk**

Small. Trivial change.

**Performance impact**

None.

**Is this a regression from a previous update?**

No.

**Root cause analysis:**

The binder is able to work properly in presence of some syntax errors, but throws for others. Our EE syntax error tests only covered syntax error that was not causing the binder to throw.

**How was the bug found?**

Customer reported.

### CODE REVIEW

**Comment by tmat** [comment]
@cston @ivanbasov @jinujoseph Please review.

File: src/ExpressionEvaluator/VisualBasic/Test/ExpressionCompiler/ExpressionCompilerTests.vb [hunk 1]
```diff
@@ -456,6 +456,11 @@ End Class
                     Assert.Null(result)
                     Assert.Equal("error BC30035: Syntax error.", errorMessage)
 
+                    '
```
Comment (cston) [inline src/ExpressionEvaluator/VisualBasic/Test/ExpressionCompiler/ExpressionCompilerTests.vb]: Missing comment?

**Comment by tmat** [comment]
test ubuntu_16_debug_prtest please

**Comment by tmat** [comment]
@Pilchie for ask mode approval.

**Comment by Pilchie** [comment]
@jinujoe @tmat we're going to have to take this to escrow, given that the final Roslyn insertion is already underway.  Please make a VSTS Bug and submit to @MattGertz.

**Comment by tmat** [comment]
@Pilchie In that case I think it's ok to move this to 15.6. This is not an actual VS crash. From the customer's point of view a "internal error in expression evaluator" message is replaced by more meaningful diagnostic message.

**Comment by Pilchie** [comment]
That sounds reasonable to me.  @jinujoseph (noticed I tagged the wrong person above.  Sorry!)

**Comment by jinujoseph** [comment]
Sounds reasonable to me as well, lets hold this till @jasonmalinowski  has a branch to take 15.6 changes

**Comment by jinujoseph** [comment]
@tmat , can you retarget this to post 15.5 branch pls

**Comment by tmat** [comment]
@MattGertz for approval

**Pull request merged**